pkg/apis/core/v1alpha1/defaults.go

```diff
@@ -235,6 +236,15 @@ func SetDefaults_Shoot(obj *Shoot) {
 	if obj.Spec.Kubernetes.KubeAPIServer.EnableAnonymousAuthentication == nil {
 		obj.Spec.Kubernetes.KubeAPIServer.EnableAnonymousAuthentication = pointer.BoolPtr(false)
 	}
+
+	if k8sVersionGreaterOrEqualThan122 {
+		for i := 0; i < len(obj.Spec.Provider.Workers); i++ {
+			if obj.Spec.Provider.Workers[i].CRI != nil {
+				continue
+			}
+			obj.Spec.Provider.Workers[i].CRI = &CRI{Name: CRINameContainerD}
+		}
+	}
```

Comment:
We can mention about the defaulting in the doc string of the corresponding field.

For example you can add:

```diff
diff --git a/pkg/apis/core/v1beta1/types_shoot.go b/pkg/apis/core/v1beta1/types_shoot.go
index 870ce8f51..4b13046de 100644
--- a/pkg/apis/core/v1beta1/types_shoot.go
+++ b/pkg/apis/core/v1beta1/types_shoot.go
@@ -980,7 +980,8 @@ type Worker struct {
        // CABundle is a certificate bundle which will be installed onto every machine of this worker pool.
        // +optional
        CABundle *string `json:"caBundle,omitempty" protobuf:"bytes,2,opt,name=caBundle"`
-       // CRI contains configurations of CRI support of every machine in the worker pool
+       // CRI contains configurations of CRI support of every machine in the worker pool.
+       // Defaults to a CRI with name "containerd" when the Kubernetes version of the Shoot is >= 1.22.
        // +optional
        CRI *CRI `json:"cri,omitempty" protobuf:"bytes,3,opt,name=cri"`
        // Kubernetes contains configuration for Kubernetes components related to this worker pool.
```

It will be then part of the API reference documentation - ref https://gardener.cloud/documentation/references/core/#core.gardener.cloud/v1beta1.Worker.